src/Orleans/Utils/WrappedException.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Orleans.Runtime.Internal
+{
+    /// <summary>Exception used to communicate with the storage provider, so that it throws this exception to its caller.</summary>
+    [Serializable]
+    public class WrappedException : OrleansException
```

Comment:
Hehe, I had exactly the same concern and went back and forth with several options. I can make this internal and add 1 InternalsVisibleTo to OrleansProviders (because this logic is spread through 3 projects)